src/mscorlib/src/System/Random.cs

```diff
@@ -114,11 +121,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
+      }
+
+      /*==================================GenerateGlobalSeed====================================
+      **Action:  Creates a number to use as global seed.
+      **Returns: An integer that is safe to use as seed values for thread-local seed generators.
+      ==========================================================================================*/
+      private static int GenerateGlobalSeed() {
+          return Guid.NewGuid().GetHashCode();
```

Comment:
Good point. What about `NewGuid() ^ TickCount`? Seems to be safer at insignificant perf cost.

Since TickCount is in practice quite structured we could even try to "whiten" the bits a bit: `x ^ RotateLeft(x, 13)`. This is a common trick that improves hash codes with just a few instructions. TickCount has many bits that are practically fixed. Rotating and xoring causes the fast-changing low order bits to make an appearance in other positions.

With all these xor tricks it is important that the bits being xored are not correlated. I see no reason whatsoever to assume that being the case here.